src/flush.rs

```diff
@@ -102,7 +107,7 @@ impl DbInner {
             wguard.pop_imm_wal();
             wguard.increment_next_wal_id();
             // flush to the memtable before notifying so that data is available for reads
-            self.flush_imm_wal_to_memtable(wguard.memtable(), imm.table());
+            self.flush_imm_wal_to_memtable(wguard.memtable(), imm.table())?;
```

Comment:
this needs to be done before `pop_imm_wal` since it can fail. Otherwise we risk returning corrupt results:
1. wal file 1 contains record foo->bar
2. flush thread flushes wal file 1
3. flush thread pops wal file 1 from imm wal table
4. flush thread fails writing foo->bar to memtable
5. flush thread releases wguard
6. reader tries to read foo->bar and gets None
7. flush thread failure handler puts db in failed state, and future reads fail, but read at step 6 succeeded and returned incorrect result

Reply:
I think this is just a bug in general, because there's no guarantee we don't panic between popping the wal and writing its data into the memtable.